### PR TITLE
Added delete function to filecenter

### PR DESCRIPTION
**Pull request by ezwang**
EDIT: test added

### CODE REVIEW

**Comment by pefoley2** [comment]
@ezwang I'd be more then happy to help you with the tests.
Feel free to join the #tjcsl-ion irc channel on freenode.net for help.
Email would work too if you prefer.

File: intranet/apps/files/views.py [hunk 1]
```diff
@@ -328,6 +328,109 @@ def can_access_path(fsdir):
...
+        if "{win}" in host_dir:
+            host_dir = windows_dir_format(host_dir, request.user)
+            try:
+                sftp.chdir(host_dir)
+            except IOError as e:
+                if "NoSuchFile" in "{}".format(e):
+                    host_dir = "/"
+                    try:
+                        sftp.chdir(host_dir)
+                    except IOError as e2:
```
Comment (pefoley2) [inline intranet/apps/files/views.py]: These nested try-catch blocks seem a bit iffy. Consider refactoring?

File: intranet/apps/files/views.py [hunk 2]
```diff
@@ -328,6 +328,109 @@ def can_access_path(fsdir):
 
 
 @login_required
+def files_delete(request, fstype=None):
+    filepath = request.GET.get("dir", None)
```
Comment (jwoglom) [inline intranet/apps/files/views.py]: This needs to use POST, not GET. GET should never be used for destructive operations.

File: intranet/apps/files/tests.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+"""
+
+
+class FilesTest(IonTestCase):
+
+    @patch('django.contrib.messages.error')
+    @patch('stat.S_ISDIR')
+    @patch('intranet.apps.files.views.get_authinfo')
+    @patch('pysftp.Connection')
+    @patch('intranet.apps.files.models.Host')
```
Comment (pefoley2) [inline intranet/apps/files/tests.py]: Rather then mocking out this, you should try and actually call the correct methods used in production.
Ideally, the only thing mocked out here should be pysftp, as it uses external resources.

File: intranet/apps/files/tests.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+        # Create hosts entry.
+        Host.objects.create(name="Computer Systems Lab",
+                            code="csl",
+                            address="remote.tjhsst.edu",
+                            linux=True)
+
+        # Login to remote file system
+        response = self.client.post(reverse('files_auth'), {'password': 'hunter2'})
+        self.assertEqual(response.status_code, 302)
```
Comment (pefoley2) [inline intranet/apps/files/tests.py]: Should this be a 302? if so, leave a comment as to why.

File: intranet/templates/files/directory.html [hunk 5]
```diff
@@ -57,29 +57,41 @@
...
+                    <a href="?dir={{ current_dir|escape }}/{{ file.name|escape }}">
                         <i class="fa fa-folder"></i>
                         {{ file.name }}
                     </a>
                 </td>
-                <td class="download-zip">
+                <td class="file-information">
+                </td>
+                <td class="file-options">
                     <a class="zip" href="?dir={{ current_dir }}/{{ file.name|escape }}&zip" data-name="{{ file.name|escape }}" onclick="return zip_folder(this)">
```
Comment (pefoley2) [inline intranet/templates/files/directory.html]: Should this & be html escaped?

**Comment by pefoley2** [comment]
Looks good, a few minor nits.
Please use git rebase -i to squash the commits as well.

File: intranet/templates/files/directory.html [hunk 6]
```diff
@@ -57,30 +57,40 @@
...
                         <i class="fa fa-folder"></i>
                         {{ file.name }}
                     </a>
                 </td>
-                <td class="download-zip">
-                    <a class="zip" href="?dir={{ current_dir }}/{{ file.name|escape }}&zip" data-name="{{ file.name|escape }}" onclick="return zip_folder(this)">
+                <td class="file-information">
+                </td>
+                <td class="file-options">
+                    <a class="zip" href="?dir={{ current_dir|escape }}/{{ file.name|escape }}&zip">
```
Comment (pefoley2) [inline intranet/templates/files/directory.html]: Should this & be html escaped?

**Comment by pefoley2** [comment]
Other then one minor nit, LGTM.

**Comment by jwoglom** [comment]
Looks good! Thanks for the test!

**Pull request merged**